app/controllers/users/fulfillment_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Users::FulfillmentController < ApplicationController
+  before_action :require_login
+
+  def index
+    my_order_items= OrderItem.where(product_id: current_user.products)
+    
+    @grouped_by_status = my_order_items.group_by { |item| item.order.order_state}
+
+    my_orders = my_order_items.pluck(:order_id)
+
+    my_orders = Order.where(id: my_orders)
+
+    if params[:order_state] && params[:order_state] != ""
+      my_orders = my_orders.where(order_state: params[:order_state])
+    end
+
```

Comment:
So, `:order_state`, this sounds like a particular order. 
But it also looks like we are selecting multiple orders that match this state.
It is possible that there is some naming convention that isn't clear or that the logic needs to be mapped out more precisely.